src/GenerateB.c

```diff
@@ -68,11 +55,23 @@ static double scale_x,scale_z; // multipliers for scaling coordinates
 static doublecomplex ki,kt;    // abs of normal components of k_inc/k0, and ktran/k0
 static doublecomplex ktVec[3]; // k_tran/k0
 static double p0;              // amplitude of the incident dipole moment
+#ifndef NO_FORTRAN
+static int n0;                 // Bessel beam order
+static double alpha0;          // half-cone angle
+static doublecomplex K,Kt,Kz;  // wave-vector components
+static doublecomplex M[4];     // Components of matrix M
```

Comment:
Please double check the behavior for complex input M. It was definitely wrong before - you should have some tests capable of detecting that.